### DISCUSSION TITLE
org.postgresql.util.PSQLException: ERROR: operator does not exist: text @ sub

### DISCUSSION THREAD

**Discussion by sitanshubhunia** [epam/Indigo#528] [Q&A]
Hi,
I am working on project that uses PostgreSQL cartridge. I am able to substructure search in query editor, but when try to run same query from java getting the error.

Query:
select * from bingo.TABLE where molecule_column @ (bingo.FileToText('D:/qrySDF.sdf'), '')::bingo.sub

Error:
org.postgresql.util.PSQLException: ERROR: operator does not exist: text @ sub
Hint: No operator matches the given name and argument type(s). You might need to add explicit type casts.

Please suggest me how to solve this problem still not find a way to overcome this problem.

**Comment by mkviatkovskii** [comment]
@AlexanderSavelyev, could you please take a look on that?

**Reply by sitanshubhunia** [reply]
@mkviatkovskii it is solved by Alexander #527 

Requesting you and your team to provide more detailed user manual for PostgreSQL(if possible). If I wants to set a rule regarding Tautomer search or any other than how to set rule on bingo.bingo_tau_config table.( please give a example and how to use this).

How to works on R-Group and how to extract descriptors. Basically we are trying to implement a system that can helps our scientist  to predict before any reaction what would be the product etc.. Actually we are trying to automate our reaction process to some extent. 

Requesting you to provide any documents or example on this , would be very beneficial for me. 
Thanks in advance.
